jsettlers.logic/src/main/java/jsettlers/logic/buildings/trading/TradingBuilding.java

```diff
@@ -119,16 +129,42 @@ public void setWaypoint(EWaypointType waypointType, ShortPoint2D position) {
 		}
 	}
 
+	private boolean coastIsReachable(ShortPoint2D position) {
+		boolean waterFound = false;
+		boolean landFound = false;
+		short x = position.x;
+		short y = position.y;
+		final byte[] xDeltaArray = EDirection.getXDeltaArray();
+		final byte[] yDeltaArray = EDirection.getYDeltaArray();
+		int distance = 5;
+		for (int direction = 0; direction < EDirection.NUMBER_OF_DIRECTIONS; direction++) {
```

Comment:
You can make this loop break faster if you change the condition to:

`direction < EDirection.NUMBER_OF_DIRECTIONS && (!waterFound || !landFound)`